app/frontend/src/features/userQueries/useUserByUsername.ts

```diff
@@ -11,15 +11,21 @@ import { service } from "service";
 import { reactQueryRetries } from "../../constants";
 
 export default function useUserByUsername(
-  username: string,
+  username: string | undefined,
   invalidate: boolean = false
 ) {
   //We look up the userId first from the username.
   //This causes a duplicate query, but it is not made stale for a long time
   //and ensures no duplication of users in the queryCache.
-  const usernameQuery = useQuery<{ username: string; userId: number }, Error>({
+  const usernameQuery = useQuery<
+    { username: string; userId: number } | null,
+    Error
+  >({
     cacheTime: username2IdStaleTime,
```

Comment:
Still feels a bit weird to allow `undefined` in the query key as they are technically pointing to "ghost entities" in the cache which feels a bit of a code smell... but I'm guessing this is the problem we've created by badly overloading our URL routes.

Maybe if `username` is undefined, we should have a super low `cacheTime` so at least React Query clears it from the cache so we don't have invalid data sitting around with something like this?
```suggestion
    cacheTime: username ? username2IdStaleTime : 0,
```

Reply:
Yes, this can be avoided with seperating the routes. Rules of hooks don't allow to call hooks conditionally so I had to allow undefined here which does look weird indeed.